test/e2e/scenario/recorder.scenario.ts

```diff
@@ -654,17 +694,305 @@ describe('recorder', () => {
         expect(segment.records.slice(3).every((record) => record.type !== RecordType.FullSnapshot)).toBe(true)
       })
   })
+
+  describe('layout viewport properties - ', () => {
+    createTest('getWindowWidth/Height reports using layout viewport dimensions')
+      .withRum({ enableExperimentalFeatures: ['visualviewport'] })
+      .withRumInit(initRumAndStartRecording)
+      .withSetup(bundleSetup)
+      .withBody(html`${VIEWPORT_META_TAGS}`)
+      .run(async ({ events }) => {
+        /**
+         * InnerWidth/Height on some devices/browsers are changed by pinch zoom
+         * We need to ensure that our measurements are not affected by pinch zoom
+         */
+        if (isGestureUnsupported()) {
+          return // No Fallback test
+        }
+        await resetViewport()
+
+        const { innerWidth, innerHeight } = (await browserExecute(() => ({
+          innerWidth: window.innerWidth,
+          innerHeight: window.innerHeight,
+        }))) as { innerWidth: number; innerHeight: number }
+
+        const initialVisualViewport = await getVisualViewport()
+        await pinchZoom(150)
+        await pinchZoom(150)
+        await sleep(210)
+        const nextVisualViewport = await getVisualViewport()
+
+        await browserExecute(() => {
+          window.dispatchEvent(new Event('resize'))
+        })
+        await sleep(210)
+
+        await flushEvents()
+        const segment = getLastSegment(events)
+        const ViewportResizeRecords = findAllIncrementalSnapshots(segment, IncrementalSource.ViewportResize)
+
+        const lastViewportResizeRecord = ViewportResizeRecords.slice(-1)[0].data as ViewportResizeData
+
+        expectToBeNearby(Math.round(lastViewportResizeRecord.width), innerWidth)
+        expectToBeNearby(Math.round(lastViewportResizeRecord.height), innerHeight)
+        // Test the test: ensure the pinch zoom worked
+        expect(initialVisualViewport.scale < nextVisualViewport.scale).toBeTruthy()
+      })
+
+    createTest('scrollX/Y reports using layout viewport dimensions')
```

Comment:
this test fails on my machine (I had to merge main to get the chrome driver update)